bindings/java/src/org/sleuthkit/datamodel/DBAccessManager.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Sleuth Kit Data Model
+ *
+ * Copyright 2018 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	 http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.logging.Logger;
+import org.sleuthkit.datamodel.SleuthkitCase.CaseDbConnection;
+import static org.sleuthkit.datamodel.SleuthkitCase.closeStatement;
+import org.sleuthkit.datamodel.TskData.DbType;
+
+/**
+ * This class provides modules with access to the case database 
+ * to create custom tables/indexes and to query them.
+ *
+ */
+public final class DBAccessManager {
+
+	private static final Logger logger = Logger.getLogger(DBAccessManager.class.getName());
+
+	private final SleuthkitCase tskDB;
+
+	/**
+	 * Constructor
+	 *
+	 * @param skCase The SleuthkitCase
+	 *
+	 */
+	DBAccessManager(SleuthkitCase skCase) {
+		this.tskDB = skCase;
+	}
+
+	/**
+	 * Creates a table with the specified name and schema.
+	 * 
+	 * @param tableName name of the table to create
+	 * @param tableSchema table schema
+	 * 
+	 * @throws TskCoreException 
+	 */
+	public void createTable(final String tableName, final String tableSchema) throws TskCoreException {
+
+		validateTableName(tableName);
+		validateSQL(tableSchema);
+
+		CaseDbConnection connection = tskDB.getConnection();
+		tskDB.acquireSingleUserCaseReadLock();
+
+		Statement statement = null;
+		String createSQL = "CREATE TABLE IF NOT EXISTS " + tableName + " " + tableSchema;
```

Comment:
Does this fail quietly if the index already exists? Is this the correct syntax for both PostgreSQL and SQLite?

What happens if there is a naming collision? We should document that tables and indexes we should be created with some unique prefix for the name.

Reply:
Correct, it "does nothing", if the table already exists.  And yes, the syntax covers both DB variants.
If there is a name collision and we don't detect and throw an exception on it ourselves, the CREATE TABLE call will do nothing.  

I will document here that clients should use their own prefixes.  Better collision detection with our core tables will be addresses later.

Reply:
Good, thanks.